MailCoreUI/Utils/FloatingPanelHelper.swift

```diff
@@ -144,17 +171,34 @@ public struct SelfSizingPanelViewModifier: ViewModifier {
         return topPadding + titleSpacing + UIFont.preferredFont(forTextStyle: .headline).pointSize
     }
 
+    private var shouldShowCloseButton: Bool {
+        return platformDetector.isMac || UIDevice.current.orientation.isLandscape
+    }
+
+    private var shouldShowZStack: Bool {
+        return title != nil || shouldShowCloseButton
+    }
+
     public init(dragIndicator: Visibility = Visibility.visible, title: String? = nil) {
         self.dragIndicator = dragIndicator
         self.title = title
     }
 
     public func body(content: Content) -> some View {
         VStack(spacing: titleSpacing) {
-            if let title {
-                Text(title)
-                    .font(Font(UIFont.preferredFont(forTextStyle: .headline)))
-                    .frame(maxWidth: .infinity)
+            if shouldShowZStack {
+                ZStack {
+                    if let title {
+                        Text(title)
+                            .font(.headline)
```

Comment:
Do not change the type of font, keep the old one: `Font(UIFont.preferredFont(forTextStyle: .headline)`.
Could you please also do that line 120, It's a mistake on my part.